app/views/posts/show.html.erb

```diff
@@ -3,35 +3,56 @@
 <div class="mt-4 border hover:border-blue-500 p-5">
   <div class="flex gap-x-28">
     <div class="">
-      <p><%= link_to @post.title, user_post_path(@post.author.id, @post) %> By <%= @post.author.name %></p>
+      <p>
+        <%= link_to @post.title, user_post_path(@post.author.id, @post) %> By
+        <%= @post.author.name %>
+      </p>
     </div>
     <div class="flex justify-end gap-x-2">
-      <p>Comments: <%= @post.comment_counter %>,</p>
+      <p>Comments: <%= @post.comments.count %>,</p>
       <p>Likes: <%= @post.like_counter %></p>
     </div>
   </div>
   <div class="mt-4">
     <p class="py-5"><%= @post.text %></p>
   </div>
   <div class="mt-4 border hover:border-blue-500 p-5">
-    <p class="underline">Comments: <%= @count %></p>
+    <p class="underline">Comments: <%= @post.comments.count %></p>
     <% @post.recent_comments.each do |comment| %>
     <div class="flex flex-col mt-2">
       <span><%= comment.user.name %> : <%= comment.text %></span>
     </div>
+    <div class="border border-slate-300 hover:border-indigo-500 cursor-pointer px-3 w-5">
+      <% if @ability.can? :delete, comment %>
+      <%= form_with model: @comment, url: user_post_comment_path(comment.post.author_id,comment.post.id,comment.id), method: :delete do |form| %>
+      <%= form.submit "Delete", class: "cursor-pointer" %>
+    <% end %>
+    <% end %>
+    </div>
     <% end %>
   </div>
-
 </div>
 <div class="mt-3 flex justify-between gap-x-40 items-center">
-  <div class="border border-slate-300 hover:border-indigo-500 cursor-pointer px-3">
-<%= form_with model: @like, url: user_post_likes_path(post_id: @post.id) do |form| %>
-  <span><%= form.submit "Like", class: "cursor-pointer" %></span>
-<% end %>
-</div>
+  <div
+    class="border border-slate-300 hover:border-indigo-500 cursor-pointer px-3"
+  >
+    <%= form_with model: @like, url: user_post_likes_path(post_id: @post.id) do
+    |form| %>
+    <span><%= form.submit "Like", class: "cursor-pointer" %></span>
+    <% end %>
+  </div>
 
+  <div
+    class="border border-slate-300 hover:border-indigo-500 cursor-pointer px-3"
+  >
+    <% if @ability.can? :delete, @post %> <%= form_with model: @post, url:
+    user_post_path(post_id: @post.id), method: :delete do |f| %> <%= f.submit
+    "Delete", class: "cursor-pointer" %> <% end %> <% end %>
+  </div>
```

Comment:
- [ ] An error occurs on the post show page as you can see below. This error prevents me from duly checking out if only authorized users can delete posts and comments. Kindly fix the issue below using the hint I provided.
![image](https://github.com/AnsarIbrahim/Bloggy/assets/46839548/f53011ed-b139-4dc0-884f-a8d89e248134)

## HINT 🔔 
Since your form is submitting to the `user_post_path`, you should specify the "author id" as well as the "post id". See the code below 👇 
```ruby
<% if @ability.can? :delete, @post %>
  <%= form_with model: @post, url: user_post_path(@post.author_id, @post.id), method: :delete do |f| %>
    <%= f.submit "Delete", class: "cursor-pointer" %>
  <% end %>
<% end %>

```

Reply:
Dear Reviewer, @codepantha

This is belong to syntaxerror template pls disable your some html extension then run it again